### PR TITLE
Updates to regexp-match-indices tests based on d-flag

### PR DESCRIPTION
**Pull request by rbuckton**
Updates based on the changes from https://github.com/tc39/proposal-regexp-match-indices/pull/49

NOTE: Its not yet clear if there are any public implementations that have adopted this change as of yet.

### CODE REVIEW

**Comment by iainireland** [comment]
Data point: I have prototyped an implementation of regexp-match-indices with the /d flag in SpiderMonkey, but we are waiting on the updated test262 tests before shipping.

It looks like V8 has [already landed their implementation with /d]( https://github.com/v8/v8/commit/81e7e2f437d20f2ecbcd7ca70c012f90e2259adb) (guarded behind a flag).

**Comment by Constellation** [comment]
https://github.com/WebKit/WebKit/commit/7e63c11fb1f6b2f5964c460a656144ee7553e604 JSC change is also ready.

**Review by msaboff — APPROVED** [review]
LGTM

File: test/built-ins/RegExp/prototype/hasIndices/cross-realm.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    1. Let R be the this value.
+    2. If Type(R) is not Object, throw a TypeError exception.
+    3. If R does not have an [[OriginalFlags]] internal slot, then
+      a. If SameValue(R, %RegExpPrototype%) is true, return undefined.
+      b. Otherwise, throw a TypeError exception.
+features: [regexp-match-indices, cross-realm]
+---*/
+
+var dotAll = Object.getOwnPropertyDescriptor(RegExp.prototype, 'hasIndices').get;
```
Comment (msaboff) [inline test/built-ins/RegExp/prototype/hasIndices/cross-realm.js:18]: I think the `dotAll` should be `hasIndices`.

File: test/built-ins/RegExp/prototype/flags/coercion-hasIndices.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+assert.sameValue(get.call(r), "", "hasIndices: null");
+
+r.hasIndices = NaN;
+assert.sameValue(get.call(r), "", "hasIndices: NaN");
+
+r.hasIndices = "";
+assert.sameValue(get.call(r), "", "hasIndices: the empty string");
+
+r.hasIndices = "string";
+assert.sameValue(get.call(r), "s", "hasIndices: string");
```
Comment (msaboff) [inline test/built-ins/RegExp/prototype/flags/coercion-hasIndices.js:33]: I think this `'s'` should be a `'d'` along with all the subsequent ones.

File: test/built-ins/RegExp/match-indices/indices-groups-properties.js [hunk 3]
```diff
@@ -7,22 +7,22 @@ includes: [compareArray.js, propertyHelper.js]
...
 assert.compareArray(groupNames, ["fst", "snd"]);
 
 // // Properties are created with Define, not Set
 // let counter = 0;
 // Object.defineProperty(Object.prototype, 'x', {set() { counter++; }});
 
-let indices = /(?<x>.)/.exec('a').indices;
+let indices = /(?<x>.)/d.exec('a').indices;
 let groups = indices.groups;
 // assert.sameValue(counter, 0);
```
Comment (msaboff) [inline test/built-ins/RegExp/match-indices/indices-groups-properties.js:27]: Is this dead code?

File: test/built-ins/RegExp/match-indices/indices-groups-properties.js [hunk 4]
```diff
@@ -7,22 +7,22 @@ includes: [compareArray.js, propertyHelper.js]
...
 
 
 // Properties created on result.groups in textual order.
-let groupNames = Object.getOwnPropertyNames(/(?<fst>.)|(?<snd>.)/u.exec("abcd").indices.groups);
+let groupNames = Object.getOwnPropertyNames(/(?<fst>.)|(?<snd>.)/du.exec("abcd").indices.groups);
 assert.compareArray(groupNames, ["fst", "snd"]);
 
 // // Properties are created with Define, not Set
 // let counter = 0;
 // Object.defineProperty(Object.prototype, 'x', {set() { counter++; }});
```
Comment (msaboff) [inline test/built-ins/RegExp/match-indices/indices-groups-properties.js:23]: Is this dead code?

Comment (rbuckton) [reply test/built-ins/RegExp/match-indices/indices-groups-properties.js:23]: Not sure why I commented it out originally. It probably should be evaluated.

**Review by msaboff — APPROVED** [review]
Looks good.  Thanks for taking care of those quickly.

**Comment by rbuckton** [comment]
FYI: I'll be putting up a PR against `engine262` for the updates as well. I've been using it to very tests locally since all of the changes fail on circleci right now.

**Comment by rbuckton** [comment]
@rwaldron can you review and (hopefully) merge?

**Comment by rbuckton** [comment]
ping: @rwaldron, @leobalter I'd appreciate a review, thanks.

**Review by rwaldron — APPROVED** [review]

**Pull request merged**